pom.xml

```diff
@@ -70,30 +70,32 @@
               <id>default-compile</id>
               <configuration>
                 <jdkToolchain>
-                  <version>9</version>
+                  <version>[1.5,9)</version>
                 </jdkToolchain>
-                <release>9</release>
+                <source>${javaVersion}</source>
+                <target>${javaVersion}</target>
+                <excludes>
+                  <exclude>module-info.java</exclude>
+                </excludes>
               </configuration>
             </execution>
             <execution>
-              <id>base-compile</id>
+              <id>java-9-compile</id>
               <goals>
                 <goal>compile</goal>
               </goals>
               <configuration>
-                <excludes>
-                  <exclude>module-info.java</exclude>
-                </excludes>
+                <jdkToolchain>
+                  <version>9</version>
+                </jdkToolchain>
+                <release>9</release>
+                <includes>
+                  <include>module-info.java</include>
+                </includes>
```

Comment:
Surprisingly this works exactly as desired:
- It sees the other source files (this is necessary for Java to check if packages referenced in `module-info` exist)
- It only compiles `module-info`